core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -477,6 +479,10 @@ protected Scope visitInsert(Insert insert, Optional<Scope> scope)
                         Joiner.on(", ").join(queryTypes));
             }
 
+            analysis.setTargetColumns(insertColumns.stream()
```

Comment:
I think this method should be called whenever `setUpdateType` is set. So I believe that adding parameter to `setUpdateType` was better option. Now it looks like it set to arbitrary list of statements.

Reply:
> I think this method should be called whenever setUpdateType

Then setUpdateType should be captured after all the validation of columns and types. 

Setting the updateType before throwing an semantic exception (current design) also allows to know what is the type of operation we were about to perform - if it is a `INSERT` or `CREATE` .. Else we might have only the exception which the user might have to parse and identify it.